src/main/java/org/opensearch/agent/tools/CreateAnomalyDetectorTool.java

```diff
@@ -0,0 +1,413 @@
+            }, e -> {
+                log.info("fail to predict model: " + e);
+                listener.onFailure(e);
+            }));
+        }, e -> {
+            log.info("fail to get mapping: " + e);
+            String errorMessage = e.getMessage();
+            if (errorMessage.contains("no such index")) {
+                listener
+                    .onFailure(
+                        new IllegalArgumentException(
+                            "Return this final answer to human directly and do not use other tools: 'Please provide index name'. Please try to directly send this message to human to ask for index name"
+                        )
+                    );
+            } else {
+                listener.onFailure(e);
+            }
+        }));
+    }
+
+    /**
+     * Enrich the parameters by adding the parameters extracted from the chat
+     * @param parameters the original parameters
+     * @return the enriched parameters with parameters extracting from the chat
+     */
+    private Map<String, String> enrichParameters(Map<String, String> parameters) {
+        Map<String, String> result = new HashMap<>(parameters);
+        try {
+            // input is a map
+            Map<String, String> chatParameters = gson.fromJson(parameters.get("input"), Map.class);
+            result.putAll(chatParameters);
+        } catch (Exception e) {
+            // input is a string
+            String indexName = parameters.getOrDefault("input", "");
+            if (!indexName.isEmpty()) {
+                result.put("index", indexName);
+            }
+        }
+        return result;
+    }
+
+    /**
+     *
+     * @param fieldsToType the flattened field-> field type mapping
+     * @return a list containing all the date type fields
+     */
+    private Set<String> findDateTypeFields(final Map<String, String> fieldsToType) {
+        Set<String> result = new HashSet<>();
+        for (Map.Entry<String, String> entry : fieldsToType.entrySet()) {
+            String value = entry.getValue();
+            if (value.equals("date") || value.equals("date_nanos")) {
+                result.add(entry.getKey());
+            }
+        }
+        return result;
+    }
+
+    /**
+     *
+     * @param fieldsToType the flattened field-> field type mapping
+     * @param indexName the index name
+     * @return the prompt about creating anomaly detector
+     */
+    private String constructPrompt(final Map<String, String> fieldsToType, final String indexName) {
+        StringJoiner tableInfoJoiner = new StringJoiner("\n");
+        for (Map.Entry<String, String> entry : fieldsToType.entrySet()) {
+            tableInfoJoiner.add("- " + entry.getKey() + ": " + entry.getValue());
+        }
+
+        return "Human:\" turn\": Here is an example of the create anomaly detector API: POST _plugins/_anomaly_detection/detectors "
```

Comment:
Can we move this prompt to a file for easier reading and reduced maintenance effort?

Reply:
You can put this into src/resources

Reply:
Sure, I'll do that.